File: app-nt-webflux/src/main/java/com/example/appntwebflux/service/NTService.java
package com.example.appntwebflux.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.MediaType;
import org.springframework.stereotype.Service;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Slf4j
@Service
@RequiredArgsConstructor
public class NTService {

    private final WebClient webClient;

    public Mono<String> test() {
        return webClient.get()
                .uri("/mock")
                .retrieve()
                .bodyToMono(String.class);
    }

    public Flux<String> stream(Integer latency, Integer length) {
        return webClient.get()
                .uri(uriStreamBuilder(latency, length))
                .accept(MediaType.TEXT_EVENT_STREAM)
                .retrieve()
                .bodyToFlux(String.class);
    }

    private String uriStreamBuilder(Integer latency, Integer length) {
        return "/stream?latency=" + latency +
                "&length=" + length;
    }

}

File: app-nt-kotlin/src/main/kotlin/org/example/appntkotlin/service/NTService.kt
package org.example.appntkotlin.service

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.IO
import kotlinx.coroutines.async
import kotlinx.coroutines.coroutineScope
import kotlinx.coroutines.withContext
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import org.springframework.web.reactive.function.client.WebClient
import kotlinx.coroutines.reactor.awaitSingle


@Service
class NTService {

    @Value("\${integration.mock.url}")
    private lateinit var url: String

    private val webClient = WebClient.create()

    suspend fun test(): String = withContext(Dispatchers.IO) {
        coroutineScope {
            val request1 = async {
                webClient.get()
                    .uri(url)
                    .retrieve()
                    .bodyToMono(String::class.java)
                    .awaitSingle()
            }

            val request2 = async {
                webClient.get()
                    .uri(url)
                    .retrieve()
                    .bodyToMono(String::class.java)
                    .awaitSingle()
            }

            val request3 = async {
                webClient.get()
                    .uri(url)
                    .retrieve()
                    .bodyToMono(String::class.java)
                    .awaitSingle()
            }

            val r1 = request1.await()
            val r2 = request2.await()
            val r3 = request3.await()

            "Results:\n$r1\n$r2\n$r3"
        }
    }
}
